ion/reader.go

```diff
@@ -227,7 +227,21 @@ func (r *reader) FieldName() *string {
 
 // Annotations returns the current value's annotations.
 func (r *reader) Annotations() []string {
-	return r.annotations
+	var annotations []string
+	for _, an := range r.annotations {
+		if an.Text != nil {
+			systemSymbolName := getSystemSymbolMapping(V1SystemSymbolTable, *an.Text)
+			if systemSymbolName != "" {
+				annotations = append(annotations, systemSymbolName)
+			} else {
+				annotations = append(annotations, *an.Text)
+			}
+		} else {
+			annotations = append(annotations, "")
```

Comment:
This code block uses the empty string as a sentinel value in several places, which worries me. Do we need to use `* string` instead of `string` to communicate the possibility of a value's absence throughout?

Reply:
I wondered the same thing but I thought we didn't want to change the reader.Annotations API?

Reply:
Opened https://github.com/amzn/ion-go/issues/151.

Reply:
Have we considered the option of an empty array instead of a 1-element array with the element being the empty string?

Reply:
> Have we considered the option of an empty array instead of a 1-element array with the element being the empty string?

How would you distinguish between an annotation we couldn't resolve and the absence of an annotation?